# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpConnection.cpp
# Commit: 1e238e8b0efa
# Full Hash: 1e238e8b0efa8e566aa0edddab4959abff427be6
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2025-01-09 09:32:25
# Description:
#   Bug 1940508 - Add null check to HandshakeDoneInternal before using mTransaction r=necko-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D233563
# ==============================================================================

diff -r 4742b9755697 -r 1e238e8b0efa netwerk/protocol/http/nsHttpConnection.cpp
--- a/netwerk/protocol/http/nsHttpConnection.cpp	Wed Jan 08 22:42:56 2025 +0000
+++ b/netwerk/protocol/http/nsHttpConnection.cpp	Wed Jan 08 22:45:14 2025 +0000
@@ -2426,7 +2426,9 @@
   if (!transactionNPN.IsEmpty() && negotiatedNPN != transactionNPN) {
     LOG(("Resetting connection due to mismatched NPN token"));
     DontReuse();
-    mTransaction->Close(NS_ERROR_NET_RESET);
+    if (mTransaction) {
+      mTransaction->Close(NS_ERROR_NET_RESET);
+    }
     return;
   }
 
